app/services/katello/pulp/repository.rb

```diff
@@ -0,0 +1,23 @@
+module Katello
+  module Pulp
+    class Repository < ::Actions::Pulp::Abstract
+      attr_accessor :repo, :input
+
+      def initialize(repo, input)
```

Comment:
input here probably doesn't make much sense, instead you probably want to pass either a smart proxy, or a pulp_resource

Reply:
We could remove the input here and just pass the Katello repository record?